src/platforms/common/server/kms-utils/kms_connector.cpp

```diff
@@ -186,20 +189,40 @@ auto mgk::find_crtc_with_primary_plane(
     {
         std::tie(crtc, crtc_index) = find_crtc_and_index_for_connector(resources, connector);
     }
-    
+
     mgk::PlaneResources plane_res{drm_fd};
 
     for (auto& plane : plane_res.planes())
     {
         if (plane->possible_crtcs & (1 << crtc_index))
         {
             ObjectProperties plane_props{drm_fd, plane->plane_id, DRM_MODE_OBJECT_PLANE};
-            if (plane_props["type"] == DRM_PLANE_TYPE_PRIMARY)
+            std::cerr << "Plane type: " << plane_props["type"] << '\n';
+        }
+    }
+
+    DRMModePlaneUPtr primary_plane, cursor_plane;
+    for (auto& plane : plane_res.planes())
+    {
+        if (plane->possible_crtcs & (1 << crtc_index))
+        {
+            ObjectProperties plane_props{drm_fd, plane->plane_id, DRM_MODE_OBJECT_PLANE};
+            switch(plane_props["type"])
             {
-                return std::make_pair(std::move(crtc), std::move(plane));
+                case DRM_PLANE_TYPE_PRIMARY:
+                    primary_plane = std::move(plane);
+                    break;
+                case DRM_PLANE_TYPE_CURSOR:
+                    cursor_plane = std::move(plane);
+                    break;
```

Comment:
Not sure if this and other cursor plane related code fit in this PR.

Reply:
Yeah, I don't think that we need any of the cursor plane code in this one. Just assume a hardware cursor exists until a failure (which should happen in the `hide()` call from the `Cursor` constructor) tells you otherwise.